common/shell/shell_platform.c

```diff
@@ -0,0 +1,607 @@
+#include <drivers/gpio.h>
+#include "plat_gpio.h"
+#include "hal_gpio.h"
+
+/* Include SENSOR */
+#include "sensor.h"
+
+/* Include config settings */
+#include "shell_platform.h"
+
+/* Declare Log */
+LOG_MODULE_REGISTER(shell_platform);
+
+/* Declare Code info */
+#define RELEASE_VERSION "v1.0.0"
+#define RELEASE_DATE "2022.01.11"
+
+/*********************************************************************************************************
+ * TOOL SECTION
+**********************************************************************************************************/
+/* 
+    Command GPIO 
+*/
+#if COMMAND_GPIO == 1
+#if PINMASK_RESERVE_CHECK
+enum CHECK_RESV {
+    CHECK_BY_GROUP_IDX,
+    CHECK_BY_GLOBAL_IDX
+};
+static int gpio_check_reserve(const struct device *dev, int gpio_idx, enum CHECK_RESV mode) 
+{
+    if (!dev)
+        return 1;
+    
+    const struct gpio_driver_config *const cfg = (const struct gpio_driver_config *)dev->config;
+
+    if (!cfg->port_pin_mask)
+        return 1;
+
+    int gpio_idx_in_group;
+    if (mode == CHECK_BY_GROUP_IDX) {
+        gpio_idx_in_group = gpio_idx;
+    } else if (mode == CHECK_BY_GLOBAL_IDX) {
+        gpio_idx_in_group = gpio_idx%GPIO_GROUP_SIZE;
+    } else
+        return 1;
+
+    if ( (cfg->port_pin_mask & (gpio_port_pins_t)BIT(gpio_idx_in_group) ) == 0U)
+        return 1;
+
+    return 0;
+}
+#endif
+
+static int gpio_access_cfg(const struct shell *shell, int gpio_idx, enum GPIO_ACCESS mode, int *data)
+{
+    if (!shell)
+        return 1;
+
+    if (gpio_idx >= GPIO_CFG_SIZE || gpio_idx < 0)
+        return 1;
+
+    /* check whether reserve pin defined in pin NAME */
+    /*
+    if ( !strncmp(GPIO_RESERVE_PREFIX, gpio_name[gpio_idx], 7) )
+        return 1;
+    */
+    switch (mode)
+    {
+    case GPIO_READ:
+        if ( gpio_cfg[gpio_idx].is_init == DISABLE )
+            return 1;
+
+        uint32_t g_val = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx/32]);
+        uint32_t g_dir = sys_read32(GPIO_GROUP_REG_ACCESS[gpio_idx/32]+0x4);
+
+        char *pin_prop = (gpio_cfg[gpio_idx].property == OPEN_DRAIN) ? "OD":"PP";
+        char *pin_dir = (gpio_cfg[gpio_idx].direction == GPIO_INPUT) ? "input":"output";
+
+        char *pin_dir_reg = "I";
+        if ( g_dir & BIT(gpio_idx%32) )
+            pin_dir_reg = "O";
+
+        int val = gpio_get(gpio_idx);
+        if (val == 0 || val == 1)
+            shell_print(shell, "[%-3d] %-35s: %-3s | %-6s(%s) | %d(%d)",
+                gpio_idx, gpio_name[gpio_idx], pin_prop, pin_dir, pin_dir_reg, val, GET_BIT_VAL(g_val, gpio_idx%32));
+        else
+            shell_print(shell, "[%-3d] %-35s: %-3s | %-6s(%s) | %s",
+                gpio_idx, gpio_name[gpio_idx], pin_prop, pin_dir, pin_dir_reg, "resv");
+        
+        break;
+
+    case GPIO_WRITE:
+        if (!data)
+            return 1;
+
+        if ( *data!=0 && *data!=1 )
+            return 1;
+
+        if ( gpio_set(gpio_idx, *data) )
+            return 1;
```

Comment:
Might be good to have a shell_print here to indicate unable to set the requested gpio.

Reply:
Got it, this suggestion would add in next pr, thanks!